drivers/adc/adc_mchp_xec_v2.c

```diff
@@ -288,11 +326,42 @@ static void adc_xec_isr(const struct device *dev)
 
 	xec_adc_get_sample(dev);
 
+#ifdef CONFIG_PM_DEVICE
+	adc_xec_pm_policy_state_lock_put(data, ADC_PM_POLICY_STATE_SINGLE_FLAG);
+#endif
+
 	adc_context_on_sampling_done(&data->ctx, dev);
 
 	LOG_DBG("ADC ISR triggered.");
 }
 
+#ifdef CONFIG_PM_DEVICE
+static int adc_xec_pm_action(const struct device *dev, enum pm_device_action action)
+{
+	const struct adc_xec_config *const devcfg = dev->config;
+	int ret;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
```

Comment:
@jvasanth1 Why are not any ADC register operations (i.e. sleep enable) 
Also, I was expecting to see additional entries in board device tree for ADC pinctrl cfg for sleep state

Reply:
@albertofloyd will add them

Reply:
@jvasanth1 which ones the pinctrl dts entries or ADC register actions? 
I understand board dts updates could be done in another PR, but if ADC registers are required they should be done in this one.

Reply:
updated both